pytorch_lightning/utilities/memory.py

```diff
@@ -20,3 +24,34 @@ def recursive_detach(in_dict: dict) -> dict:
         else:
             out_dict.update({k: v})
     return out_dict
+
+
+def is_oom_error(exception):
+    return is_cuda_out_of_memory(exception) \
+        or is_cudnn_snafu(exception) \
+        or is_out_of_cpu_memory(exception)
+
+
+def is_cuda_out_of_memory(exception):
+    return isinstance(exception, RuntimeError) \
+        and len(exception.args) == 1 \
+        and "CUDA out of memory." in exception.args[0]
+
+
+def is_cudnn_snafu(exception):
+    return isinstance(exception, RuntimeError) \
+        and len(exception.args) == 1 \
+        and "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED." in exception.args[0]
```

Comment:
Thanks for this PR to implement toma in PyTorch Lightning!

If you copy code and ideas from my projects, could you please add a mention from it, too? I see that you're a fellow PhD student, so you are aware of the importance of credit assignment.

In particular, if you copy code verbatim and remove helpful comments... maybe add them back.

```
def is_cuda_out_of_memory(exception):
    return (
        isinstance(exception, RuntimeError) and len(exception.args) == 1 and "CUDA out of memory." in exception.args[0]
    )


def is_cudnn_snafu(exception):
    # For/because of https://github.com/pytorch/pytorch/issues/4107
    return (
        isinstance(exception, RuntimeError)
        and len(exception.args) == 1
        and "cuDNN error: CUDNN_STATUS_NOT_SUPPORTED." in exception.args[0]
    )
```

```
def gc_cuda():
    """Gargage collect Torch (CUDA) memory."""
    gc.collect()
    if torch.cuda.is_available():
        torch.cuda.empty_cache()
```

is from https://github.com/BlackHC/toma/blob/master/toma/torch_cuda_memory.py.

Now, I think this PR contains lots of other code, and I think it's great, but maybe add a link or a mention.

Thank you,
 Andreas